test/functional/test-experiments.js

```diff
@@ -793,3 +796,182 @@ describe('experiment branch tests', () => {
+        'JuM/w25iNA6Xo1mEMQDjuASrJheVErl3KRbZTi4++sbpY3Six2jyxUZlHgjQgYaz9e9/' +
+        'wP4RGg0a3FMHzKKNpmzFY7tTcsPbgeZ1EPkz4suyjl6vL2zkbDUgO/5GKD9A5oQNS6w1' +
+        '3NN7';
+
+    // Same as correctToken except Version: 47
+    tokenWithBadVersion =
+        'LwAAAJF7Im9yaWdpbiI6Imh0dHBzOi8vd3d3Lmdvb2dsZS5jb20iLCJleHBlcmltZW50' +
+        'cyI6eyJhbXAtZXhwaXJlcy1sYXRlciI6eyJleHBpcmF0aW9uIjo5NTYxNzYwMjAwMDAw' +
+        'MH0sImFtcC1leHBpcmVkLSI6eyJleHBpcmF0aW9uIjoxMjMyNDI3NjAwMDAwfX19AC0Q' +
+        'XglvsXoxh8tpQN1/ZAQm/2c6O8JZwdcAylHhOn0tA6RiYtaQMycgqrvBOWIPWO10Nalm' +
+        '4xvi6BNPZRJCpVDsj4Bt8tCr2n6kZWuA0a5MqQTfpEZfTBLLB8UPlryGHXYSIHrtZahb' +
+        'AyQMs/AzLp/IotDsUEhPWJg5ApjnM1uw4djIq2lAUrlc4a8mM8s4i0h4Ung3RFkgmott' +
+        'JuM/w25iNA6Xo1mEMQDjuASrJheVErl3KRbZTi4++sbpY3Six2jyxUZlHgjQgYaz9e9/' +
+        'wP4RGg0a3FMHzKKNpmzFY7tTcsPbgeZ1EPkz4suyjl6vL2zkbDUgO/5GKD9A5oQNS6w1' +
+        '3NN7';
+    // Same as correctToken except length = 999999
+    tokenWithBadConfigLength =
+        'AAAPQj97Im9yaWdpbiI6Imh0dHBzOi8vd3d3Lmdvb2dsZS5jb20iLCJleHBlcmltZW50' +
+        'cyI6eyJhbXAtZXhwaXJlcy1sYXRlciI6eyJleHBpcmF0aW9uIjo5NTYxNzYwMjAwMDAw' +
+        'MH0sImFtcC1leHBpcmVkLSI6eyJleHBpcmF0aW9uIjoxMjMyNDI3NjAwMDAwfX19AC0Q' +
+        'XglvsXoxh8tpQN1/ZAQm/2c6O8JZwdcAylHhOn0tA6RiYtaQMycgqrvBOWIPWO10Nalm' +
+        '4xvi6BNPZRJCpVDsj4Bt8tCr2n6kZWuA0a5MqQTfpEZfTBLLB8UPlryGHXYSIHrtZahb' +
+        'AyQMs/AzLp/IotDsUEhPWJg5ApjnM1uw4djIq2lAUrlc4a8mM8s4i0h4Ung3RFkgmott' +
+        'JuM/w25iNA6Xo1mEMQDjuASrJheVErl3KRbZTi4++sbpY3Six2jyxUZlHgjQgYaz9e9/' +
+        'wP4RGg0a3FMHzKKNpmzFY7tTcsPbgeZ1EPkz4suyjl6vL2zkbDUgO/5GKD9A5oQNS6w1' +
+        '3NN7';
+     // Same as correctToken except with a bad signature
+    tokenWithBadSignature =
+        'AAAAAJF7Im9yaWdpbiI6Imh0dHBzOi8vd3d3Lmdvb2dsZS5jb20iLCJleHBlcmltZW50' +
+        'cyI6eyJhbXAtZXhwaXJlcy1sYXRlciI6eyJleHBpcmF0aW9uIjo5NTYxNzYwMjAwMDAw' +
+        'MH0sImFtcC1leHBpcmVkLSI6eyJleHBpcmF0aW9uIjoxMjMyNDI3NjAwMDAwfX19YmFk' +
+        'c2lnbmF0dXJlISEhISEh';
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  function setupMetaTagWith(token) {
+    const meta = document.createElement('meta');
+    meta.setAttribute('name', 'amp-experiment-token');
+    if (token) {
+      meta.setAttribute('content', token);
+    }
+    win.document.head.appendChild(meta);
+  }
+
+  it('should return false if no token is found', () => {
+    // No meta tag ever added
+    const p = isExperimentOnForOriginTrial(win, 'amp-expires-later', publicJwk);
+    return expect(p).to.eventually.be.false;
+  });
+
+  it('should return false if crypto is unavailable', () => {
+    sandbox.stub(crypto, 'isCryptoAvailable').returns(false);
+    const p = isExperimentOnForOriginTrial(win, 'amp-expires-later', publicJwk);
+    return expect(p).to.eventually.be.false;
+  });
+
+  it('should throw for missing token', () => {
+    if (!crypto.isCryptoAvailable()) { return; }
```

Comment:
Extract this to wrap `it` methods instead? Probably better to not run than appear to pass.